packages/rainbowkit/src/components/RainbowKitProvider/AuthenticationContext.tsx

```diff
@@ -89,11 +90,11 @@ export function RainbowKitAuthenticationProvider<Message = unknown>({
   useEffect(() => {
     if (connector && status === 'authenticated') {
       // Attach the event listener when status is 'authenticated'
-      connector.on('change', handleChangedAccount);
+      connector.emitter.on('change', handleChangedAccount);
 
       // Cleanup function to remove the event listener
       return () => {
-        connector?.off('change', handleChangedAccount);
+        connector.emitter.off('change', handleChangedAccount);
       };
     }
```

Comment:
Took a look at this approach more closely and tried to find some built in functions / hooks for these emitters but seems like it's only built for `onConnect` and `onDisconnect`. For now we can just pull in the `emitter` from the connector.